packages/relay/src/lib/eth.ts

```diff
@@ -1567,16 +1566,30 @@ export class EthImpl implements Eth {
     let fileId: FileId | null = null;
     let txSubmitted = false;
     try {
-      const sendRawTransactionResult = await this.hapiService
-        .getSDKClient()
-        .submitEthereumTransaction(
-          transactionBuffer,
-          EthImpl.ethSendRawTransaction,
-          requestDetails,
-          originalCallerAddress,
-          networkGasPriceInWeiBars,
-          await this.getCurrentNetworkExchangeRateInCents(requestDetails),
-        );
+      const sendRawTransactionResult = await this.sendRawTransactionWithRetry(
+        async () =>
+          await this.hapiService
+            .getSDKClient()
+            .submitEthereumTransaction(
+              transactionBuffer,
+              EthImpl.ethSendRawTransaction,
+              requestDetails,
+              originalCallerAddress,
+              networkGasPriceInWeiBars,
+              await this.getCurrentNetworkExchangeRateInCents(requestDetails),
+            ),
+        {
+          canRetry: (e: unknown) => {
+            return e instanceof SDKClientError && (e.isConnectionDropped() || e.isTimeoutExceeded());
+          },
```

Comment:
To make sure we don't duplicate transactions when a timeout occurs, we can get the `transactionId` from the `SDKClientError`: https://github.com/hashgraph/hedera-sdk-js/blob/15cb9c91209dd68667e138f191b606c04d73a8d6/src/transaction/TransactionResponse.js#L87 and poll the mirror node to confirm the transaction has actually failed before retrying:

**MirrorNodeClient.ts:**
```typescript
public async isFailedTransaction(transactionId: string, requestDetails: RequestDetails): Promise<boolean> {
  const transactionDetails = await this.repeatedRequest(
    this.getTransactionById.name,
    [transactionId, requestDetails],
    this.MIRROR_NODE_REQUEST_RETRY_COUNT,
    requestDetails,
  );
  return transactionDetails[0]?.result !== 'SUCCESS';
}
```
**eth.ts**:
```suggestion
          canRetry: async (e: unknown) => {
            return e instanceof SDKClientError && (e.isConnectionDropped() || e.isTimeoutExceeded()) && await this.mirrorNodeClient.isFailedTransaction(e.transactionId);
          },
```

Reply:
However, what you have linked is the ReceiptStatusError and in a timeout GrpcServiceError is thrown https://github.com/hashgraph/hedera-sdk-js/blob/main/src/Executable.js#L662
But I will find a way to get the transacion id anyways